llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -2342,6 +2343,174 @@ SmallVector<utils::IteratorType> UnPackOp::getLoopIteratorTypes() {
   return iteratorTypes;
 }
 
+//===----------------------------------------------------------------------===//
+// WinogradInputTransformOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult WinogradInputTransformOp::verify() {
+  Operation *op = getOperation();
+  if (getNumInputs() != 1) {
+    return op->emitOpError("expected one input operand");
+  }
+  if (getNumOutputs() != 1) {
+    return op->emitOpError("expected one output operand");
+  }
+  if (!input().getType().isa<ShapedType>()) {
+    return op->emitOpError("expected input type to be shaped");
+  }
+  if (!output().getType().isa<ShapedType>()) {
+    return op->emitOpError("expected output type to be shaped");
+  }
+  auto inputType = input().getType().cast<ShapedType>();
+  auto outputType = output().getType().cast<ShapedType>();
+  ArrayRef<int64_t> inputShape = inputType.getShape();
+  if (inputShape.size() != 4) {
+    return op->emitOpError("expected input operand to have rank 4");
+  }
+  ArrayRef<int64_t> outputShape = outputType.getShape();
+  if (outputType.getElementType() != inputType.getElementType()) {
+    return op->emitOpError(
+        "expected input/output element types to be identical");
+  }
+  if (getOutputOperandRank() != getInputOperandRank() + 2) {
+    return op->emitOpError(
+        "expected output rank to be equal to input rank + 2");
+  }
+  // Check shapes (assumes incoming input shape in NHWC)
+  const int64_t outputTileSize = getOutputTileSize();
+  const int64_t kernelSize = getKernelSize();
+  const int64_t inputTileSize = getInputTileSize();
+  SmallVector<int64_t> expectedOutputShape(getInputOperandRank() + 2,
+                                           inputTileSize);
+  for (int i = 0; i < inputShape.size(); i++) {
+    if ((i == 0) || (i == 3) || (inputShape[i] == ShapedType::kDynamicSize)) {
```

Comment:
Dont understand the dynamic size check. Also FYI, this is now renamed to `ShapedType::kDynamic`

Reply:
So the idea is that for dynamic shapes if the input shape is dynamic then we allow the output shape to be dynamic as well. Thanks will have to rebase and update to use ShapedType::kDynamic.

Reply:
yes you are right, areShapesCompatible should work here. Will remove the dynamic check.